src/lib/core/CHIPConfig.h

```diff
@@ -1667,6 +1667,22 @@ extern const char CHIP_NON_PRODUCTION_MARKER[];
 #define CHIP_CONFIG_ICD_CLIENTS_SUPPORTED_PER_FABRIC 2
 #endif
 
+/**
+ * @def CHIP_CONFIG_ICD_MAX_CLIENTS
+ *
+ * @brief
+ *   Maximum number of ICD clients. Based on this number, platforms that utilize the
+ *   PSA Crypto API should reserve key slot range.
+ *
+ * @note
+ *   For platforms that utilize the PSA Crypto API, this configuration is used to
+ *   compute the number of PSA key slots. It should remain unchanged during the device's lifetime,
+ *   as alterations may lead to issues with backwards compatibility.
+ */
+#ifndef CHIP_CONFIG_ICD_MAX_CLIENTS
```

Comment:
nit: consider making this config specifically for the PSA crypto, such as `CHIP_CONFIG_CRYPTO_PSA_ICD_MAX_CLIENTS` as if it's made too generic, it may be mistakenly used in other contexts where it's assumed to mean the current maximum, not maximum maximum :D.

Reply:
makes sense :) renamed